files/usr/share/jeos-firstboot/jeos-firstboot-functions

```diff
@@ -27,15 +27,28 @@ list=
 password=''
 modules=()
 
-if pushd "${jeos_prefix?}/share/jeos-firstboot/modules" &>/dev/null; then
-        for module in *; do
-		# shellcheck source=/dev/null
-                if [ -f "${module}" ] && source "${module}"; then
-                        modules+=("${module}")
-                fi
-        done
-        popd &>/dev/null || exit
-fi
+_find_modules() {
+	local d f
+	for d in "/etc/jeos-firstboot/modules" "${jeos_prefix?}/share/jeos-firstboot/modules"; do
+		for f in "$d"/*; do
+			if [ -L "$f" ] || [ -f "$f" ]; then
+				echo "${f##*/}" "$f"
+			fi
+		done
+	done
+}
+
+init_modules() {
+	[ -z "$modules" ] || return 0
+	local module f
+	while read -r module f; do
+		if [ -L "$f" ] && [ "$(readlink "$f")" = "/dev/null" ]; then
+			continue
+		fi
+		. "$f"
```

Comment:
Needs to do `source "$f"` within an if to not kill the entire process if a module has a syntax error